src/audio/audio.hpp

```diff
@@ -105,6 +105,19 @@ class Audio
 	 * @see Seek
 	 */
 	virtual std::uint64_t Position() const = 0;
+
+	/**
+	 * This Audio's length.
+	 * @return Audio's length, in microseconds.
+	 */
+	virtual std::uint64_t TotalLength() const = 0;
```

Comment:
Something for later might be figuring out if this should be an `int64_t`, given how other things may be using signed integers only.